src/EditorFeatures/VisualBasic/UseAutoProperty/UseAutoPropertyAnalyzer.vb

```diff
@@ -0,0 +1,138 @@
+            Return Utilities.GetNodeToRemove(identifier)
+        End Function
+
+        Protected Overrides Function IsEligibleHeuristic(field As IFieldSymbol, propertyDeclaration As PropertyBlockSyntax, compilation As Compilation, cancellationToken As CancellationToken) As Boolean
+            If propertyDeclaration.Accessors.Any(SyntaxKind.SetAccessorBlock) Then
+                ' If this property already has a setter, then we can definitely simplify it to an auto-prop 
+                Return True
+            End If
+
+            ' the property doesn't have a setter currently. check all the types the field is 
+            ' declared in.  If the field is written to outside of a constructor, then this 
+            ' field Is Not elegible for replacement with an auto prop.  We'd have to make 
+            ' the autoprop read/write, And that could be opening up the propert widely 
+            ' (in accessibility terms) in a way the user would not want.
+            Dim containingType = field.ContainingType
+            For Each ref In containingType.DeclaringSyntaxReferences
+                Dim containingNode = ref.GetSyntax(cancellationToken)
+                If containingNode IsNot Nothing Then
+                    Dim semanticModel = compilation.GetSemanticModel(containingNode.SyntaxTree)
+                    If IsWrittenOutsideOfConstructorOrProperty(field, propertyDeclaration, containingNode, semanticModel, cancellationToken) Then
+                        Return False
+                    End If
+                End If
+            Next
+
+            ' No problem simplifying this field.
+            Return True
+        End Function
+
+        Private Function IsWrittenOutsideOfConstructorOrProperty(field As IFieldSymbol,
+                                                                 propertyDeclaration As PropertyBlockSyntax,
+                                                                 node As SyntaxNode,
+                                                                 semanticModel As SemanticModel,
+                                                                 cancellationToken As CancellationToken) As Boolean
+            cancellationToken.ThrowIfCancellationRequested()
+
+            If node Is propertyDeclaration Then
+                Return False
+            End If
+
+            If node.Kind() = SyntaxKind.ConstructorBlock Then
+                Return False
+            End If
+
+            If node.Kind() = SyntaxKind.IdentifierName Then
+                Dim symbol = semanticModel.GetSymbolInfo(node)
+                If field.Equals(symbol.Symbol) Then
+                    If semanticFacts.IsWrittenTo(semanticModel, node, cancellationToken) Then
+                        Return True
+                    End If
+                End If
+            Else
+                For Each child In node.ChildNodesAndTokens
```

Comment:
That's an enumerable.  I wanted to avoid the allocations.  ChildNodesAndTokens doesn't allocate.